### ISSUE TITLE
Data submitters want to order the datasets in a collection based on their relevance

### ISSUE THREAD

**Issue by brianraymor** [chanzuckerberg/single-cell#159]
**Story**

*Data submitters want to order datasets in the collection detail and datasets view based on their relevance.*

**Background**

@pablo-gar recently curated [Tabula Muris Senis](https://cellxgene.cziscience.com/collections/0b9d8a04-bb9d-44da-aa27-705bb65b54eb) which contains 42 datasets. Pablo wanted to ensure that the most **relevant** datasets were at the top to be easily discovered by consumers.  The most relevant - both the _All — A single-cell transcriptomic atlas characterizes ageing tissues in the mouse_ datasets are currently listed at the bottom making it harder for consumers to discover.

There is another case that is pending publication:

> ... the most relevant dataset is the one where data from multiple papers was integrated into a single dataset.The others are either subsets of that dataset or the original datasets from the papers. That integrated dataset is currently in the middle.

**May 2022**

Request from the Allen Institute to define the precise order of datasets in a collection.

---

Research from @tihuan on potential FE solutions:

> Ooh I just did a quick search, looks like HTML5 has [drag and drop API](https://caniuse.com/dragndrop)
> And [this article](https://dev.to/colinmcd01/drag-drop-re-ordering-using-html-and-react-974) details how to do it with React! 
> So maybe we just need a quick POC to make sure it works with an HTML table too, then we’re good?

**Comment by brianraymor** [comment]
Closing. This has not been a priority for more than 2 years. This would be expensive to implement due to the increased complexity in the Curation UX, Discover UX, Portal API, and data store to maintain dataset ordering per collection. 

I noted a simpler approach above :

> One solution could be a sort by column hint set by curators for the collection detail page.



CC: @jahilton  @hthomas-czi

**Comment by danieljhegeman** [comment]
At present, the Portal API returns datasets without an explicitly-determined order; presumably this is being governed by default by the date of creation, or howsoever psql decides to yield the query results.

A couple concerns: the main concern appears to be ordering "by importance", which is obviously subjective (rather than some objective measure like date revised, title, etc.). To me the only reasonable approach for satisfying this feature request is to **provide the curators the ability to manually drag/reorder the Datasets in the UI** (in the private/revision state, of course -- is a revision that only results in reordering of Datasets truly a revision? Open question...).

We'd either...
1) Add an explicit positioning index column on table `DatasetVersion`, or
2) Express order through the order of their uuids in the `CollectionVersion.datasets` array.

**Comment by brianraymor** [comment]
> To me the only reasonable approach for satisfying this feature request is to provide the curators the ability to manually drag/reorder the Datasets in the UI

Yes. That's why I recommended that CC t-shirt size this epic in its entirety. 

> presumably this is being governed by default by the date of creation, or howsoever psql decides to yield the query results.

The UX sorts datasets in a collection by descending order of cell count.

**Comment by metakuni** [comment]
> Yes. That's why I recommended that CC t-shirt size this epic in its entirety.

Oops, sorry, @brianraymor , I either missed or forgot that and asked @danieljhegeman to estimate the BE part.
